filament/backend/src/vulkan/VulkanTexture.cpp

```diff
@@ -245,12 +245,14 @@ void VulkanTexture::updateWithCopyBuffer(const PixelBufferDescriptor& hostData,
 
     const VkCommandBuffer cmdbuffer = mContext.commands->get().cmdbuffer;
 
-    // We can't blindly use LAYOUT_UNDEFINED because it may destroy the data, and because
-    // we're potentially updating only a sub-region it would be a problem.
+    // We may use LAYOUT_UNDEFINED if we can tell the update is not partial.
+    // Besides, the newly created texture is LAYOUT_UNDEFINED, and thus an error will occur from
+    // the khronos validation layer if we blindly use the LAYOUT_SHADER_READ_ONLY_OPTIMAL.
+    bool fullupdate = (width == this->width) && (height == this->height) && (depth == this->depth);
     VkImageLayout textureLayout = mContext.getTextureLayout(usage);
     transitionImageLayout(cmdbuffer, textureTransitionHelper({
             .image = mTextureImage,
-            .oldLayout = textureLayout,
+            .oldLayout = fullupdate ? VK_IMAGE_LAYOUT_UNDEFINED : textureLayout,
```

Comment:
It seems to me that the problem is that `textureLayout` is not initialized properly somewhere else.
This fix is more like a workaround.

Reply:
Actually, we need some additional information to indicate whether the texture is newly created. However, the additional information may make the program not efficient.

Reply:
Yes. This patch doesn't totally fix this bug. However, this patch may boost performance since we use the LAYOUT_UNDEFINED.